[lihongguang00] iP

lihongguang00 opened the pull request:
# ChadGPT
> I love ChadGPT - Bill gates

ChadGPT is your new go-to assistant to organise your day. Say goodbye to ~~messy calenders~~ and say hello to an organised one. It's
+ clean
+ effective

All you need to do is:

1. download it from [here](https://github.com/lihongguang00/ip)
2. double-click it
3. add your tasks
4. let it manage your task for you 👍 

And it is **FREE**

Features
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

# 

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

lihongguang00 committed:
Level-0a

Renamed Chatbot

lihongguang00 committed:
Level-0b

Reformatted horizontal lines, added goodbye statement

lihongguang00 committed:
Level-1

ChadGPT echoes commands entered by users, until user enters 'bye'

lihongguang00 committed:
Level-2

Added list feature

lihongguang00 committed:
Level-3

Added feature that allows tasks to be marked as done and undone

lihongguang00 committed:
Level-4

Added ToDo, Deadline, Event tasks, claned up code in main function of Duke.java

lihongguang00 committed:
Level-5

Included exception handling, added helper function to query status of task in task.java

lihongguang00 committed:
Level-6

Implemented delete function with relevant exception handling

lihongguang00 committed:
Level-7 v1.0

Under construction

lihongguang00 committed:
Level-7 v2.0

Add task list memory feature

lihongguang00 committed:
Level-8 v1.0

Work in progress

lihongguang00 committed:
Level-8 v2.0

In progress

lihongguang00 committed:
Level-8 v3.0

Add feature where time and date are stored as LocalDate and LocalTime objects

lihongguang00 committed:
A-MoreOOP v1.0

Work in progress

lihongguang00 committed:
A-MoreOOP v2.0

Work-in-progress

lihongguang00 committed:
A-MoreOOP v3.0

work in progress, added more classes

wjayee commented on src/main/java/dukeexceptions/CorruptDataException.java:
```diff
@@ -0,0 +1,14 @@
+package dukeexceptions;
```
I like how you put all your classes in packages !

wjayee commented on src/main/java/tasks/Event.java:
```diff
@@ -0,0 +1,76 @@
+        }
+    }
+
+    private String getStartDateTimeForPrinting() {
```
I like the naming of your methods (use of verbs)

wjayee commented on src/main/java/tasks/TaskAbstract.java:
```diff
@@ -0,0 +1,51 @@
+
+public abstract class TaskAbstract {
+    protected String description;
+    protected boolean isDone;
```
I like the naming of your boolean isDone

wjayee commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,361 @@
+
+    private static TaskAbstract createTask(String nextLine, String[] strArr) {
+        String[] delimited = nextLine.split("/");
+        switch(strArr[0].toLowerCase()) {
```
Perhaps the indentation for case can be decreased by 4 (aligned with switch)

felibunnyy commented on src/main/java/tasks/Event.java:
```diff
@@ -0,0 +1,76 @@
+
+    public Event(String description, String start, String end) {
+        super(description);
+        String[] startDateArr = start.split(" ");
```
naming is all done in camelCase and is very clean and neat

felibunnyy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,361 @@
+import tasks.Deadline;
+import tasks.Event;
+import dukeexceptions.CorruptDataException;
+
```
good line breaks and indentations to make code readable

felibunnyy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,361 @@
+                    return false;
+                }
+
+                try {
```
might want to include finally statement for try catch block

felibunnyy reviewed (commented):
very clean code with most logical units separated clearly :)

lihongguang00 committed:
A-MoreOOP v4.0

Work in progress

lihongguang00 committed:
A-MoreOOP v5.0

Separated into appropriate classes, cleaned up code.

lihongguang00 committed:
Merge branch 'branch-A-MoreOOP'

lihongguang00 committed:
A-Packages v1.0

Fixed minor bugs and organised into packages

lihongguang00 committed:
Merge branch 'branch-A-Packages'

lihongguang00 committed:
Merge branch 'add-gradle-support'

lihongguang00 committed:
A-JUnit

Added JUnit test for todo and event JUnit testing

lihongguang00 committed:
Merge branch 'branch-A-JUnit'

lihongguang00 committed:
A-JavaDoc

Added JavaDocs to all classes

lihongguang00 committed:
A-CodingStandard

Ensure code is in accordance with Google java coding standard

lihongguang00 committed:
Level-9 v1.0

Added find function and fixed minor codingstyle errors

lihongguang00 committed:
Merge branch 'branch-A-JavaDoc'

lihongguang00 committed:
Merge branch 'branch-A-CodingStandard'

lihongguang00 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/assets/commands/ByeCommand.java
#	src/main/java/duke/assets/commands/CreateDeadlineCommand.java
#	src/main/java/duke/assets/commands/ListCommand.java
#	src/main/java/duke/assets/commands/OperationOnListCommandAbstract.java
#	src/main/java/duke/assets/parser/Parser.java
#	src/main/java/duke/assets/storage/Storage.java
#	src/main/java/duke/assets/storage/TaskList.java
#	src/main/java/duke/assets/tasks/Deadline.java
#	src/main/java/duke/assets/tasks/Event.java

lihongguang00 committed:
A-CheckStyle

Fixed all coding style violations with checkstyle

lihongguang00 committed:
Merge branch 'branch-A-CheckStyle'

lihongguang00 committed:
Level-10

Added JavaFX GUI

lihongguang00 committed:
Merge branch 'branch-Level-10'

lihongguang00 committed:
Level-10 v2.0

Fix bug where GUI would not show in the case where data is corrupted.

GUI will now ask user to restart the application as well as highlight the problematic part of the data file

lihongguang00 committed:
branch-A-Assertions v1.0

Unable to find major assumptions at the moment, will attempt later

lihongguang00 committed:
Merge branch 'branch-A-Assertions'

lihongguang00 committed:
A-CodeQuality

Tidy up code and removed unused methods

Added more JavaDocs and refined others

lihongguang00 committed:
Merge branch 'branch-A-CodeQuality'

lihongguang00 committed:
A-CodeQuality

Tidy up code and removed unused methods

Added more JavaDocs and refined others

lihongguang00 committed:
A-Assertions

Add assert(false) statements in methods that are assumed to be never called

lihongguang00 committed:
Merge pull request #1 from lihongguang00/branch-A-Assertions

A-Assertions

lihongguang00 committed:
Merge branch 'branch-A-CodeQuality'

lihongguang00 committed:
A-Streams

Use Java stream to replace for loop

lihongguang00 committed:
Merge branch 'branch-A-Streams'

lihongguang00 committed:
C-Sort

Add feature where users can now sort their tasklist either in ascneding/descending lexicographical or chronological order

lihongguang00 committed:
Merge branch 'branch-C-Sort'

lihongguang00 committed:
A-UserGuide

Fix minor bug and added more exception handling

Add more JUnit Testing

Add user guide

lihongguang00 committed:
Merge branch 'branch-A-UserGuide'

lihongguang00 committed:
A-UserGuide v2.0

Fixed formatting error in user guide

lihongguang00 committed:
Merge branch 'branch-A-UserGuide'

lihongguang00 committed:
A-UserGuide v3.0

Fixed Table formatting

lihongguang00 committed:
Merge branch 'branch-A-UserGuide'

lihongguang00 committed:
A-UserGuide v4.0

Fixed Table formatting

lihongguang00 committed:
Merge branch 'branch-A-UserGuide'

lihongguang00 committed:
UI image update

Updated UI image

lihongguang00 committed:
More J-Unit Tests

Add more comprehensive and rigorous JUnit testing

lihongguang00 committed:
Add Greeting

Add greeting by chatbot

lihongguang00 committed:
Bug fixes

Minor bug fixes
